nuitka/nodes/ListNodes.py

```diff
@@ -0,0 +1,317 @@
+)
+from .NodeBases import SideEffectsFromChildrenMixin
+from .NodeMetaClasses import NodeCheckMetaClass
+
+
+class ExpressionListOperationAppend(
+    ExpressionListShapeExactMixin, ExpressionChildHavingBase
+):
+    """This operation represents l.append(object)."""
+
+    kind = "EXPRESSION_LIST_OPERATION_APPEND"
+
+    name_children = ('list_arg', 'item')
+
+    def __init__(self, list_arg, item, source_ref):
+        assert list_arg is not None
+        assert item is not None
+
+
+        ExpressionChildHavingBase.__init__(
+            self,
+            value={'list_arg': list_arg, 'item': item},
+            source_ref=source_ref,
+        )
+
+    def computeExpression(self, trace_collection):
+        list_arg = self.subnode_list_arg
+        item = self.subnode_item
+
+        trace_collection.removeKnowledge(list_arg)
+
+        return self, None, None
+
+
+class ExpressionListOperationClear(
+    ExpressionListShapeExactMixin, ExpressionChildHavingBase
+):
+    """This operation represents l.clear()."""
+
+    kind = "EXPRESSION_LIST_OPERATION_CLEAR"
+
+    name_children = ('list_arg',)
+
+    def __init__(self, list_arg, source_ref):
+        assert list_arg is not None
+
+        ExpressionChildHavingBase.__init__(
+            self,
+            value=list_arg,
+            source_ref=source_ref,
+        )
+
+    def computeExpression(self, trace_collection):
+        list_arg = self.subnode_list_arg
+
+        trace_collection.removeKnowledge(list_arg)
+
+        return self, None, None
+
+
+class ExpressionListOperationCopy(
+    ExpressionListShapeExactMixin, ExpressionChildHavingBase
+):
+    """This operation represents l.copy()."""
+
+    kind = "EXPRESSION_LIST_OPERATION_COPY"
+
+    name_children = ('list_arg',)
+
+    def __init__(self, list_arg, source_ref):
+        assert list_arg is not None
+
+        ExpressionChildHavingBase.__init__(
+            self,
+            value=list_arg,
+            source_ref=source_ref,
+        )
+
+    def computeExpression(self, trace_collection):
+        return self, None, None
+
+
+class ExpressionListOperationExtend(
+    ExpressionListShapeExactMixin, ExpressionChildHavingBase
+):
+    """This operation represents l.extend(iterable)."""
+
+    kind = "EXPRESSION_LIST_OPERATION_EXTEND"
+
+    name_children = ('list_arg', 'value')
+
+    def __init__(self, list_arg, value, source_ref):
+        assert list_arg is not None
+        assert value is not None
+
+        ExpressionChildHavingBase.__init__(
+            self,
+            value={'list_arg': list_arg, 'value': value},
+            source_ref=source_ref,
+        )
+
+    def computeExpression(self, trace_collection):
+        list_arg = self.subnode_list_arg
+        value = self.subnode_value
+
+        trace_collection.removeKnowledge(list_arg)
+
+        trace_collection.onExceptionRaiseExit(BaseException)
+
+        return self, None, None
+
+
+class ExpressionListOperationIndex(
```

Comment:
This has `int` result shape. It's what an operation returns. Most operations on containers return `None` because they are pseudo-expressions, but not all.